p2p/module.go

```diff
@@ -31,38 +32,69 @@ type p2pModule struct {
 	address  cryptoPocket.Address
 
 	network typesP2P.Network
-}
 
-// TECHDEBT(drewsky): Discuss how to best expose/access `Address` throughout the codebase.
-func (m *p2pModule) GetAddress() (cryptoPocket.Address, error) {
-	return m.address, nil
+	injectedAddrBookProvider      providers.AddrBookProvider
+	injectedCurrentHeightProvider providers.CurrentHeightProvider
 }
 
 func Create(runtimeMgr modules.RuntimeMgr) (modules.Module, error) {
 	return new(p2pModule).Create(runtimeMgr)
 }
 
-func (*p2pModule) Create(runtimeMgr modules.RuntimeMgr) (modules.Module, error) {
+// IMPROVE: need to define a better pattern for dependency injection. Currently we are probably limiting ourselves by having a common constructor `Create(runtimeMgr modules.RuntimeMgr) (modules.Module, error)` for all modules.
+func CreateWithProviders(runtimeMgr modules.RuntimeMgr, addrBookProvider providers.AddrBookProvider, currentHeightProvider providers.CurrentHeightProvider) (modules.Module, error) {
```

Comment:
I left similar comments elsewhere but this might requires some more discussion, but I'm not a big fan of `CreateWithProviders` and the evolution this is taking. It's beginning to be hard to understand.

Let's think about how we can potentially fix it in this PR. For context, I opened up #84 a long time ago but closed it as "won't do" because I think our module + bus + mockgen infra should be enough

**Ideas**:
1. Ideally `runtimeMgr` should be able to host everything it needs and "be" the DI framework.
2. For "debug" providers, I think we can just use "mockgen" and potentially avoid the extra code while making it very explicit what debug is.

**Things to look into:**

1. I'm looking at `getCurrentHeightProvider` and `getAddrBookProvider` below, and ideally we shouldn't need those functions at all. When you prepare `runtimeMgr` and the bus, we should be able to inject real/mocked versions of the modules and use those elsewhere.
2. I'm looking at `debug_*.go` and think we could use mockgen instead.

Note that I did not sit and try to implement this myself, so it might get ugly or have other issues, but was wondering if you could look into it and lmk what you think of it as a potential (simpler?) approach.

Reply:
😍 It makes **total** sense and I moved into that direction in my other PR #425 for this exact reason.
With that we'll have a generic module registration on the bus that will basically be a simple DI framework. At that point this `CreateWithProviders` can be refactored as a simple `Create` and internally source these dependencies from the bus.

Please take a quick look at the bus, at the modules `Create` in #425 and at how I changed/simplified/genericized the registration pattern. 

If you like what you see I suppose I can mark this `CreateWithProvider` here as `//HACK` ~~IMPROVE~~ that can be removed in #425 or a new specific issue/PR (up to you)